codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -1178,44 +1227,38 @@ mod test {
-                                                 decodedEvent.body->Js.Array2.\
-                                                 unsafe_get(3)->HyperSyncClient.Decoder.\
-                                                 toUnderlying->Utils.magic,\n      }\n    }"
-                .to_string(),
-            decode_hyper_fuel_data_code: "(_) => Js.Exn.raiseError(\"HyperFuel decoder not \
-                                          implemented\")"
-                .to_string(),
-            params_raw_event_schema:
-                "S.object((s): eventArgs => {id: s.field(\"id\", BigInt.schema), owner: \
-                               s.field(\"owner\", Address.schema), displayName: \
-                               s.field(\"displayName\", S.string), imageUrl: \
-                               s.field(\"imageUrl\", S.string)})"
-                    .to_string(),
-            get_topic_selection_code: "(eventFilters) => \
-                                       eventFilters->SingleOrMultiple.normalizeOrThrow->Belt.\
-                                       Array.map(_eventFilter => \
-                                       LogSelection.makeTopicSelection(~topic0=[sighash->EvmTypes.\
-                                       Hex.fromStringUnsafe], )->Utils.unwrapResultExn)"
-                .to_string(),
-            event_filter_type: "{  }".to_string(),
+            module_code: format!(
+                r#"
+let sighash = "{sighash}"
+let topicCount = 1
+let name = "NewGravatar"
+let contractName = contractName
+
+@genType
+type eventArgs = {{id: bigint, owner: Address.t, displayName: string, imageUrl: string}}
+let paramsRawEventSchema = S.object((s): eventArgs => {{id: s.field("id", BigInt.schema), owner: s.field("owner", Address.schema), displayName: s.field("displayName", S.string), imageUrl: s.field("imageUrl", S.string)}})
+let convertHyperSyncEventArgs = (decodedEvent: HyperSyncClient.Decoder.decodedEvent): eventArgs => {{
+      {{
+        id: decodedEvent.body->Js.Array2.unsafe_get(0)->HyperSyncClient.Decoder.toUnderlying->Utils.magic,
+        owner: decodedEvent.body->Js.Array2.unsafe_get(1)->HyperSyncClient.Decoder.toUnderlying->Utils.magic,
+        displayName: decodedEvent.body->Js.Array2.unsafe_get(2)->HyperSyncClient.Decoder.toUnderlying->Utils.magic,
+        imageUrl: decodedEvent.body->Js.Array2.unsafe_get(3)->HyperSyncClient.Decoder.toUnderlying->Utils.magic,
+      }}
+    }}
+
+let handlerRegister: HandlerTypes.Register.t<eventArgs> = HandlerTypes.Register.make(
+  ~topic0=sighash->EvmTypes.Hex.fromStringUnsafe,
+  ~contractName,
+  ~eventName=name,
+)
+
+@genType
+type eventFilter = {{  }}
+
+let getTopicSelection = (eventFilters) => eventFilters->SingleOrMultiple.normalizeOrThrow->Belt.Array.map(_eventFilter => LogSelection.makeTopicSelection(~topic0=[sighash->EvmTypes.Hex.fromStringUnsafe], )->Utils.unwrapResultExn)"#
+            ),
```

Comment:
I would prefer to model this whole "Event module" as a struct with display implemented to format this string. Then testing can be much easier to read and write etc.